onnx_tf/backend.py

```diff
@@ -517,23 +517,24 @@ def handle_batch_normalization(cls, node, input_dict):
     total_num_dim = len(x.get_shape())
     scale = cls._explicit_broadcast(input_dict[node.inputs[1]], 1, total_num_dim)
     bias = cls._explicit_broadcast(input_dict[node.inputs[2]], 1, total_num_dim)
-    mean = cls._explicit_broadcast(input_dict[node.inputs[3]], 1, total_num_dim)
-    variance = cls._explicit_broadcast(input_dict[node.inputs[4]], 1, total_num_dim)
+    running_mean = cls._explicit_broadcast(input_dict[node.inputs[3]], 1, total_num_dim)
+    running_variance = cls._explicit_broadcast(input_dict[node.inputs[4]], 1, total_num_dim)
 
     variance_epsilon = node.attrs.get("epsilon", 0.00001)
     if node.attrs.get("is_test", 0):
-      return [tf.nn.batch_normalization(x, mean, variance, bias, scale,
+      return [tf.nn.batch_normalization(x, running_mean, running_variance, bias, scale,
                                         variance_epsilon)]
-    if "momentum" in node.attrs.keys():
-      warnings.warn("Unsupported momentum attribute by Tensorflow in "
-                    "batch_normalization. This attribute will be ignored.",
-                    UserWarning)
-    if "spatial" in node.attrs.keys():
-      warnings.warn("Unsupported spatial attribute by Tensorflow in "
-                    "batch_normalization. This attribute will be ignored.",
-                    UserWarning)
+    spatial = node.attrs.get("spatial", 1) == 1
+    momentum = node.attrs.get("momentum", 0.9)
+    axis = [0] if spatial else [0] + list(range(2, total_num_dim))
+    mean, variance = tf.nn.moments(x, axis)
+    for i in axis:
```

Comment:
Just wondering why did you not use explicit_broadcast here?

Reply:
@tjingrant 
I found it does not work in some cases. So I write by myself here.
But I fix `explicit_broadcast` by another pr finally.
https://github.com/onnx/onnx-tensorflow/pull/62/files#diff-f565871850f10b4c435ec99014386b77L182

Reply:
@tjingrant 
So I prefer you to review that one first, if merged, I will add a fix to this pr.